### ISSUE TITLE
Bug: "Type mismatch, expected text" in properties field

### ISSUE THREAD

**Issue by andresgongora** [SkepticMystic/breadcrumbs#503]
**Describe the bug**
Breadcrumbs writes the properties as `down: ['My File']` instead of `down: "[[My File]]"`, causing Obsidian to create a small warning with the message `Type mismatch, expected text`. The name of the referred file in the property does not appear as a link either, but breadcrumbs seems to work otherwise as expected and can be navigated in the matrix view.

**To Reproduce**
In a new Obsidian vault, with only breadcrumbs installed, and a default configuration where we enable writing all relationships to the files. 

Given two files `My Parent File` and `My Child file`, if I set the relationship of both files by writing only to the child the following:
```
---
up: "[[My Parent File]]"
---
```
and then ask breadcrumbs to regenerate the indices and write to all files, the content of the parent file is written as:
```
---
up: ['My Child File']
---
```

**Expected behavior**
Links are written to the property field in the `"[[Name of file]]"` format instead, such that obsidian recognizes it as a working link. In the above example, I would have expected `up: "[[My Child File]]"` instead of `up: ['My Child File']`.

**Screenshots**
![image](https://github.com/SkepticMystic/breadcrumbs/assets/12051510/9eaeac42-4341-4613-91aa-659223334f0a)

**Desktop (please complete the following information):**
 - OS: Manjaro (Linux)
 - Obsidian 1.5.3
 - Breadcrumbs 3.6.11

**Additional context**
After some more experimenting, if the files are located in folders and subfolders, writing the hierarchy can become quite mangled. I believe this can be because BC likely checks if the files are already written in the hierarchy, but because the strings `"[[File]]"` and `['File']` are not the same, it writes it twice. However, this bug might be unrelated and warrant its own issue ticket. 
![image](https://github.com/SkepticMystic/breadcrumbs/assets/12051510/b91fa077-6fc2-4593-8f7a-3cbc1609a627)
This is from another vault. The blue links I added manually and are of the type `"[[File]]"` , the white links are the ones added by BC and are of the type `['File']`

On the other hand, I'm completely new to Obsidian. I've searched thoroughly for this bug, but maybe I don't know the correct keywords to identify it and google yields no useful results.

**Comment by SkepticMystic** [comment]
Hey, thanks for the detailed issue!
I'm rewriting the plugin from scratch, please see the `V4.md` file in the root of the repo for more info.
A couple of the issues you mentioned here are resolved in the new version

**Comment by SkepticMystic** [comment]
Of course feel free to test the beta version using BRAT, and let me know your thoughts :)
Keep it open for now. I'll go through and close out any resolved issues once v4 is live.

**Comment by andresgongora** [comment]
I'm playing around with v4 and it seems to do exactly what I expected from it. I haven't had time to go in-depth but it looks promising. 

I'm migrating an extremely large wiki (believe me, it's VERY big, at least in terms of file count) from TiddlyWiki into Obsidian. I'll give feedback if I experience any slow-downs or related performance issues. So far so good! :)

**Comment by SkepticMystic** [comment]
That's great to hear!
The code should be alot more performant now as well. I don't know how many notes you've got, but it should handle them better

**Comment by andresgongora** [comment]
I can confirm that with 1100 .MDs its working nicely :D

Sorry that I hijack this thread. Can you confirm that _Folders can be used as the source of edges_? I can't find where to enable this option but it shows as done in `v4.md`. If it should indeed be enabled, I'll open a new ticket with a full report :)

**Comment by andresgongora** [comment]
Great! I can confirm it works! :D
I haven't found a way to refresh + freeze all pages at once, so I'm doing so as I go, but so far v4 works like a charm

**Comment by SkepticMystic** [comment]
That's one of the breaking changes, it's now `BC-folder-note-field`.
If you open up the Obsidian console, then rebuild the BC graph, you should see a log about `explicit edge sources`, or something similar. If you open that up, you should see errors for Folder Notes telling you about the breaking change (but there's no big notification letting you know, it's pretty hidden)

**Comment by andresgongora** [comment]
Oh no! This is my bad. I read you, then went on with my usual stuff, and when I came back to this i had a nagging feeling I was forgetting something about how the property was spelled (hence my question). Thanks! I can confirm that it does indeed work :)

As for giving a special meaning to the name of the file, maybe it's now even needed if the user sets up a template. Once everything is in place, there is no need to recurse downwards

**Comment by SkepticMystic** [comment]
Please clarify your point about folder notes and templates, I don't follow

**Comment by andresgongora** [comment]
> Please clarify your point about folder notes and templates, I don't follow

I think some of the plugin that handle folders, such as `AidenLx's Folder Note`, have an option* to define the default content of new _Folder Pages_ when they are added (ie a template for new folder notes). Meaning that I could add `BC-folder-note-field: down` to this template and incorporate it seamlessly into my workflow

*At least I believe I've seen this template option somewhere. Maybe it was a different Folder Note plugin

**Comment by SkepticMystic** [comment]
Oh cool! Yeah, that integrates really well, thanks for sharing.
Gonna point to this issue #504 for when I update the docs on this feature

**Comment by SkepticMystic** [comment]
Yeah, nice :)

Indeed, the `content` isn't implemented yet!

**Comment by SkepticMystic** [comment]
I'm gonna close out the original issue as complete. V4 isn't fully live, but is quite stable at this point. Thanks again for all the detail! Please do share any V4 feedback, it's really helpul :)
